botocore/parsers.py

```diff
@@ -386,10 +386,13 @@ def _do_error_parse(self, response, shape):
         root = self._parse_xml_string_to_dom(xml_contents)
         parsed = self._build_name_to_xml_node(root)
         self._replace_nodes(parsed)
-        # Once we've converted xml->dict, we need to make one more
-        # adjustment to be consistent with ResponseMetadata
-        # for non-error responses:
-        # {"RequestId": "id"} -> {"ResponseMetadata": {"RequestId": "id"}}
+        # Once we've converted xml->dict, we need to make one or two
+        # more adjustments to extract nested errors and to be consistent
+        # with ResponseMetadata for non-error responses:
+        # 1. {"Errors": {"Error": {...}}} -> {"Error": {...}}
+        # 2. {"RequestId": "id"} -> {"ResponseMetadata": {"RequestId": "id"}}
+        if 'Errors' in parsed and isinstance(parsed['Errors'], dict):
```

Comment:
@jamesls I've added another test in `test_parsers.py` that covers the SDB case. Since we do not control the response I was checking for a dict to ensure that we never call `parsed.update(...)` with a non-dictionary type. This prevents awkward and potentially confusing type or value errors if a service ever returns something weird.